lib/src/utils/models/file_info.dart

```diff
@@ -5,13 +5,22 @@ import 'package:equatable/equatable.dart';
 import 'package:matrix/matrix.dart';
 import 'package:mime/mime.dart';
 
+typedef ProgressCallback = void Function(int count, int total);
+
 class FileInfo with EquatableMixin {
   final String fileName;
   final String filePath;
   final int fileSize;
   final Stream<List<int>>? readStream;
+  final ProgressCallback? progressCallback;
```

Comment:
what kind of progress? @sherlockvn

Reply:
and why FileInfo need to have the progress info?

Reply:
yeah, i think it better to be inside mixin, so that later upload/download can have progress

Reply:
it's progress for upload/download file

Reply:
remove it